roachprod: remove special cases for defunct versions

RaduBerinde opened the pull request:
We now run roachprod from the same branch with the release we are
testing. We may use roachprod with the previous major version for
mixed-version testing.

This change removes any code specific to versions before 21.1. We
still want to support 21.1 because this change is intended to get
backported to release-21.2.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72958)
<!-- Reviewable:end -->

rail reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>
Wheeeeeeee! (code--)++

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72958#-:-MosilJIAyN6ME3wK09y:begpa59)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @healthy-pod)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
roachprod: remove special cases for defunct versions

We now run roachprod from the same branch with the release we are
testing. We may use roachprod with the previous major version for
mixed-version testing.

This change removes any code specific to versions before 21.1. We
still want to support 21.1 because this change is intended to get
backported to release-21.2.

Release note: None

RaduBerinde commented:
TFTR!

bors r+
